Library/Homebrew/bottles.rb

```diff
@@ -47,10 +47,10 @@ def bottle_resolve_formula_names(bottle_file)
   name = receipt_file_path.split("/").first
   tap = Tab.from_file_content(receipt_file, "#{bottle_file}/#{receipt_file_path}").tap
 
-  if tap.nil? || tap == "Homebrew/homebrew"
+  if tap.nil? || tap.is_a?(CoreTap)
```

Comment:
How about teaching `Tap` itself whether it is the `CoreFormulaTap`? There could be a `Tap#core?` method that checks the tap name or has different implementations for the two tap classes.